python/ray/data/impl/lazy_block_list.py

```diff
@@ -1,18 +1,25 @@
 import math
-from typing import Callable, List, Iterator, Tuple
+from typing import List, Iterator, Tuple, Optional, Dict, Any
+import uuid
 
 import numpy as np
 
 import ray
 from ray.types import ObjectRef
 from ray.data.block import (
     Block,
+    BlockAccessor,
+    BlockExecStats,
     BlockMetadata,
     BlockPartitionMetadata,
     MaybeBlockPartition,
 )
 from ray.data.context import DatasetContext
+from ray.data.datasource import ReadTask
 from ray.data.impl.block_list import BlockList
+from ray.data.impl.progress_bar import ProgressBar
+from ray.data.impl.remote_fn import cached_remote_fn
+from ray.data.impl.stats import DatasetStats, _get_or_create_stats_actor
 
 
 class LazyBlockList(BlockList):
```

Comment:
Can you add some unit tests for the public methods of this class?

Reply:
I'm happy to do this as a follow up, but since we don't currently have any unit testing at this layer (all unit testing is done on the `Dataset` and `DatasetPipeline` methods), I'd prefer not to block out-of-band serialization from landing on adding a new testing layer.

Reply:
I think in generally we should strive for covering the code with unit test in the same PR, rather than followups, which is essentially a (short-term) tech debt; especially for newly added classes/functions, or new public methods in existing class. This PR is mostly refactoring, and we still jam unit tests together so far, so LG to move forward.

Reply:
Strongly agree, I'm not suggesting that we should add new abstractions and the corresponding unit tests in separate PRs, I'm pushing for not making a large test coverage change in this PR for a refactoring of an _existing_ abstraction that's blocking some AIR features. I would really like to have a much more strict unit testing standard, since most of our Python tests in core Ray and our ecosystem are technically integration tests, with many intermediate edge cases uncovered. So I'm very happy that you're of the same mind, definitely looking forward to pushing on this!